src/app/app.component.ts

```diff
@@ -197,21 +203,22 @@ export class AppComponent {
         });
 
         this.tasks = Array.from(taskMap.values());
-      },
-      error: (error: any) => {
+      }),
+      catchError((error: any) => {
         console.error('Error loading tasks:', error);
         this.snackBar.open('Error loading data', 'Close', {
           duration: 3000,
         });
-      },
-    });
+        return [];
+      }),
+    );
   }
 
   loadSubtasks(maintaskId: string): void {
     this.taskService
       .loadSubtasks(maintaskId)
       .then((subtasksObservable: any) => {
-        subtasksObservable.subscribe({
+        subtasksObservable.pipe(take(1)).subscribe({
```

Comment:
Note that .subscribe() is different from .then():
An Observable could trigger subscribe many times. A Promise will only resolve once. If you want to just get something for once from an Observable, use take(1).